integration-tests/rest-client-reactive/src/main/java/io/quarkus/it/rest/client/main/ParamConverter.java

```diff
@@ -0,0 +1,46 @@
+package io.quarkus.it.rest.client.main;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+
+import javax.ws.rs.ext.ParamConverterProvider;
+
+public class ParamConverter implements ParamConverterProvider {
+    @SuppressWarnings("unchecked")
+    @Override
+    public <T> javax.ws.rs.ext.ParamConverter<T> getConverter(Class<T> rawType, Type genericType,
+            Annotation[] annotations) {
+        if (genericType == null) {
```

Comment:
it would be worth checking some actual values here, I mean adding some annotations and checking if they are passed properly, and checking the generic type with generics

Reply:
+1 
Added